lib/client.js

```diff
@@ -143,6 +155,30 @@ extend(Raven.prototype, {
     return this;
   },
 
+  makeErrorHandler: function () {
+    var self = this;
+    var calledUncaughtHandler = false;
+    var calledFatalError = false;
+    var firstError;
+    return function (err) {
+      if (!calledUncaughtHandler) {
+        firstError = err;
+        calledUncaughtHandler = true;
```

Comment:
would you consider getting rid of calledUncaughtHandler and using firstError being truthy instead? could be less to juggle here.